src/vumi2/transports/smpp/client.py

```diff
@@ -246,14 +238,12 @@ async def send_pdu(
             await self.pdu_send_channel.send(pdu)
             return None
 
-        send_channel: MemorySendChannel
-        receive_channel: MemoryReceiveChannel
-        send_channel, receive_channel = open_memory_channel(0)
-        self.responses[pdu.seqNum] = send_channel
+        self.send_channel, self.receive_channel = open_memory_channel(0)
+        self.responses[pdu.seqNum] = self.send_channel
 
         await self.pdu_send_channel.send(pdu)
 
-        async for response in receive_channel:
+        async for response in self.receive_channel:
```

Comment:
Why do `send_channel` and `receive_channel` need to become instance attributes instead of local variables here?